### PR TITLE
Rewrite

### PR DESCRIPTION
**Pull request by revolgy-rud**

### CODE REVIEW

**Review by zurekarol — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -1,267 +1,650 @@
...
+
+> Terraform is really a relatively simple tool which through its providers is capable of provisioning infrastructure on almost any cloud platform with an API.
+> On each run of Terraform, it checks the entirety of the current state and code base, which only gets slower with each new resource, and Terraform doesn't have a built-in solution to this problem, other than that there is a simple recommendation to keep its code bases small and in stages.
+> Terraform must be able to resolve the dependency graph as it's written in code, otherwise it won't run.
+
+### Strategies
+
+At Revolgy, we have elected to use the following procedures, processes, or strategies.
+
+- [Trunk Based Development]
```
[the remark this answers is not in the document]
Comment (revolgy-rud) [reply README.md:26]: Hmm, I think we need @MichalReznickyRevolgy for that, as I didn't choose it.

File: README.md [hunk 2]
```diff
@@ -1,267 +1,650 @@
...
-
-**New environment variables:**
-- env var : env var Staging variable for Rules
-
-**New scripts:**
-- script : script details
-
-**New dependencies:**
-- dependency : dependency details
+# Revolgy Development Manifesto
```
[the remark this answers is not in the document]
Comment (revolgy-rud) [reply README.md:1]: I would rather not, we would have to think of a tool to generate it. It should be a part of the viewer, and GitHub actually has it.
image

File: README.md [hunk 3]
```diff
@@ -1,267 +1,650 @@
...
+
+> **Examples**. Examples of using the module should exist under the `examples/` subdirectory at the root of the repository. Each example may have a README to explain the goal and usage of the example. Examples for submodules should also be placed in the root `examples/` directory.
+> 
+> Because examples will often be copied into other repositories for customization, any `module` blocks should have their `source` set to the address an external caller would use, not to a relative path.
+
+### `README.md`
+
+**Every** Terraform code base, regardless of whether it is a project or a module, should have a README.
+
+Use the [terraform-docs] command to generate the strictly technical parts of it. This process can (and should) be automated. [Link](https://github.com/terraform-docs/terraform-docs/blob/master/docs/USER_GUIDE.md#integrating-with-your-terraform-repository)
```
[the remark this answers is not in the document]
Comment (revolgy-rud) [reply README.md]: Please elaborate, what do you mean?

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (MichalReznickyRevolgy) [reply README.md:26]: here's the reasoning, but might be obsolete now: https://www.notion.so/revolgy/Branching-strategy-guide-f3c79363bfe643289b0874d61b3c0b6a#66c55d5084e048d6a5a27d1a244b5ca1

[the remark this answers is not in the document]
Comment (revolgy-rud) [reply README.md:26]: Honestly I'll just add my own, what's written there isn't even motivation, much less reasoning.

**Pull request merged**